fix hardcoded compiler names in Makefile + don't try to link binaries in /usr/local/bin
author: Kenneth Hoste (HPC-UGent), ported by Maxime Boissonneault (ComputeCanada)
author: fenglai liu (fenglai@accre.vanderbilt.edu)
updated for newer version by: Pavel Grochal (INUITS)
diff Augustus-3.3.3/auxprogs/bam2wig/Makefile{.orig,} -ru
--- Augustus-3.3.3/auxprogs/bam2wig/Makefile.orig	2020-02-28 14:17:14.932644416 +0100
+++ Augustus-3.3.3/auxprogs/bam2wig/Makefile	2020-02-28 14:19:06.313789625 +0100
@@ -15,14 +15,14 @@
 PROGRAM = bam2wig
 SOURCES = $(PROGRAM)
 OBJECTS = $(SOURCES:.c=.o)
-SAMTOOLS=$(TOOLDIR)/samtools
-HTSLIB=$(TOOLDIR)/htslib
-BCFTOOLS=$(TOOLDIR)/bcftools
-INCLUDES=-I$(SAMTOOLS) -I. -I$(HTSLIB) -I$(BCFTOOLS)
+SAMTOOLS=$(EBROOTSAMTOOLS)
+HTSLIB=$(EBROOTHTSLIB)
+# BCFTOOLS=$(TOOLDIR)/bcftools
+INCLUDES=-I$(SAMTOOLS)/include/bam -I. -I$(HTSLIB)/include
 VPATH=$(SAMTOOLS)
-LIBS=$(SAMTOOLS)/libbam.a $(HTSLIB)/libhts.a -lcurses -lm -lz -lpthread -lcurl -lssl -lcrypto
-CFLAGS=-Wall -O2 $(INCLUDES)
-CC=gcc
+LIBS=$(SAMTOOLS)/lib/libbam.a $(HTSLIB)/lib/libhts.a -lcurses -lm -lz -lpthread -lcurl -lssl -lcrypto
+CFLAGS:=-Wall -O2 $(CFLAGS) $(INCLUDES)
+CC:=$(CC)
 
 $(PROGRAM) : bam2wig.o
 	$(CC) $(CFLAGS) $^ -o $@ $(LIBS) -lbz2 -llzma
diff Augustus-3.3.3/auxprogs/compileSpliceCands/Makefile{.orig,} -ru
--- Augustus-3.3.3/auxprogs/compileSpliceCands/Makefile.orig	2020-02-28 15:13:11.439232201 +0100
+++ Augustus-3.3.3/auxprogs/compileSpliceCands/Makefile	2020-02-28 15:14:06.494123252 +0100
@@ -1,8 +1,8 @@
 compileSpliceCands : compileSpliceCands.o list.h list.o
-	gcc $(LDFLAGS) -o compileSpliceCands compileSpliceCands.o list.o;
+	$(CC) $(LDFLAGS) -o compileSpliceCands compileSpliceCands.o list.o;
 #	cp compileSpliceCands ../../bin
-compileSpliceCands.o : compileSpliceCands.c 
-	gcc -Wall -pedantic -ansi $(CPPFLAGS) -c compileSpliceCands.c 
+compileSpliceCands.o : compileSpliceCands.c
+	$(CC) -Wall -pedantic -ansi $(CPPFLAGS) -c compileSpliceCands.c
 
 all : compileSpliceCands
 
diff Augustus-3.3.3/auxprogs/homGeneMapping/src/Makefile{.orig,} -ru
--- Augustus-3.3.3/auxprogs/homGeneMapping/src/Makefile.orig	2020-02-28 14:24:41.294099314 +0100
+++ Augustus-3.3.3/auxprogs/homGeneMapping/src/Makefile	2020-02-28 14:24:58.937714359 +0100
@@ -7,7 +7,7 @@
 # database access for retrieval of hints
 # SQLITE = true
 
-CC      = g++
+CC      = $(CXX)
 
 # Notes: - "-Wno-sign-compare" eliminates a high number of warnings (see footnote below). Please adopt
 #          a strict signed-only usage strategy to avoid mistakes since we are not warned about this.
diff Augustus-3.3.3/Makefile{.orig,} -ru
--- Augustus-3.3.3/Makefile.orig	2020-02-28 14:27:07.166947473 +0100
+++ Augustus-3.3.3/Makefile	2020-02-28 14:27:39.330260218 +0100
@@ -17,13 +17,13 @@
 install:
 	install -d $(INSTALLDIR)
 	cp -a config bin scripts $(INSTALLDIR)
-	ln -sf $(INSTALLDIR)/bin/augustus /usr/local/bin/augustus
-	ln -sf $(INSTALLDIR)/bin/etraining /usr/local/bin/etraining
-	ln -sf $(INSTALLDIR)/bin/prepareAlign /usr/local/bin/prepareAlign
-	ln -sf $(INSTALLDIR)/bin/fastBlockSearch /usr/local/bin/fastBlockSearch
-	ln -sf $(INSTALLDIR)/bin/load2db /usr/local/bin/load2db
-	ln -sf $(INSTALLDIR)/bin/getSeq /usr/local/bin/getSeq
-	ln -sf $(INSTALLDIR)/bin/espoca /usr/local/bin/espoca
+	# ln -sf $(INSTALLDIR)/bin/augustus /usr/local/bin/augustus
+	# ln -sf $(INSTALLDIR)/bin/etraining /usr/local/bin/etraining
+	# ln -sf $(INSTALLDIR)/bin/prepareAlign /usr/local/bin/prepareAlign
+	# ln -sf $(INSTALLDIR)/bin/fastBlockSearch /usr/local/bin/fastBlockSearch
+	# ln -sf $(INSTALLDIR)/bin/load2db /usr/local/bin/load2db
+	# ln -sf $(INSTALLDIR)/bin/getSeq /usr/local/bin/getSeq
+	# ln -sf $(INSTALLDIR)/bin/espoca /usr/local/bin/espoca
 
 # for internal purposes:
 release:
diff Augustus-3.3.3/auxprogs/bam2hints/Makefile{.orig,} -ru
--- Augustus-3.3.3/auxprogs/bam2hints/Makefile.orig	2020-02-28 15:21:20.913321499 +0100
+++ Augustus-3.3.3/auxprogs/bam2hints/Makefile	2020-02-28 15:24:25.481563967 +0100
@@ -8,13 +8,13 @@
 #	Last modified:  09-October-2015 by Katharina J. Hoff
 
 # Variable definition
-INCLUDES = /usr/include/bamtools
+INCLUDES = $(EBROOTBAMTOOLS)/include/bamtools
 LIBS = -lbamtools -lz
 SOURCES = bam2hints.cc 
 OBJECTS = $(SOURCES:.cc=.o)
 CXXFLAGS += -Wall -O2 # -g -p -g -ggdb 
 
-LINK.cc = g++
+LINK.cc = $(CXX)
 
 # Recipe(s)
 # $@: full target name of current target. 
diff Augustus-3.3.3/auxprogs/filterBam/src/Makefile{.orig,} -ru
--- Augustus-3.3.3/auxprogs/filterBam/src/Makefile.orig	2020-02-28 14:41:32.389683743 +0100
+++ Augustus-3.3.3/auxprogs/filterBam/src/Makefile	2020-02-28 14:43:12.219652446 +0100
@@ -8,7 +8,7 @@
 			printElapsedTime.cc sumMandIOperations.cc sumDandIOperations.cc PairednessCoverage.cc
 PROGRAM = filterBam
 OBJECTS = $(SOURCES:.cc=.o)
-BAMTOOLS = /usr/include/bamtools
+BAMTOOLS = $(EBROOTBAMTOOLS)/include/bamtools
 INCLUDES = -I$(BAMTOOLS) -Iheaders -I./bamtools
 LIBS = -lbamtools -lz
 CFLAGS = -std=c++0x
